### DISCUSSION TITLE
Commits on branches should count towards profile contributions

### DISCUSSION THREAD

**Discussion by tammymogul** [community/community#65920] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

This has been an ongoing challenge with Github for 10+ years. I've historically had teams that do a lot of research work on branches, which often never get merged to the main branch and that's good and expected. However when you look at those profiles, you would think they haven't been doing any work at all. 

I think the policy of only counting contributions when they are merged into the main branch needs to be reconsidered. Work on a branch is still work, and it should be counted.

**Comment by hades255** [comment]
I think that is not necessary.
When you done pull request, perhaps delete that branch.
So why branch history needed?

**Comment by KashanSiddiqui** [comment]
Agree on the problem, there have been alot of devs working on branches for different needs and we often deploy branch on subdomains while team continue to work for major features on main branch. By the looks of graph it shows the devs never worked in that time.

**Comment by pantharshit007** [comment]
I am not sure but from what I know commits from branches (even your own repos) when merged should be counted as n commits not 1 merge commit.

**Comment by ktsreddy007** [comment]
Hi @tammymogul  

If branch is temporary which we delete once after merged so i think we should not take branch into consideration , Instead i think they need to take PR as consideration like how many valid PR will be triggered by Dev and based on approved PR's count only the contributions count should be taken into consideration what do you say about it ???.

**Comment by luisd-telus** [comment]
I think that branches should get taken into account as a separate measurement and brought in to the the main activity board on merge to main branch maybe yellow (or blinking green) for commits that don't go into main and then get replaced retroactively by solid green after the merge/pr. That way employers can easily see work done and work merged, sometimes we can have months of feature development in a branch and go to prod months later. If management knew how to get to those graphs, we would get fired for "not working" while the graph would look like a Christmas tree if we counted branch commits.

**Reply by chandhru2008** [reply]
I think what he said was correct

**Comment by pcal43** [comment]
+1.  Another obvious use case is commits on maintenance branches.  Backports are contributions, too.

**Comment by MattLP** [comment]
GitLab does it very well, I'm disapointed that GitHub does not track contributions on all branches...

**Comment by HamzaMhirsi** [comment]
I agree, I've been doing a lot of work on a different branch for days, and then it just gets logged when I push to main. I would really appreciate it if that gets solved.

A tip: Otherwise, I made a script to commit nothing to a random main project every day I commit something to any branch xD

**Reply by einar-hjortdal** [reply]
Share that with me please :rofl: I've been working mainly in secondary branches for months

**Reply by pantharshit007** [reply]
You can ask any agent to write a script which monitors commits on lets say `~/personal/xyz` so now on every commit where `branch != default || main || master` replicate a similar commit (can just copy the message or just the hash itself) on a different private repo.

haven't tried it, but that tip gave me this idea.

**Reply by einar-hjortdal** [reply]
I do not have any agent bro.
I used your suggestion and wrote the script myself.
```
#!/bin/bash

COMMIT_MSG=$(git log -1 --pretty=%B) # sync commit message
cd ~/Documents/projects/activity-sync || exit # path of private repository
git commit --allow-empty -m "Sync: $COMMIT_MSG"
git push origin
```
Put this in your repositories `.git/hooks` I think you can figure out the full step-by-step
